packages/webapp/components/layouts/AccountLayout/Profile/VerifiedCompanyBadge/VerifiedCompanyBadgeSection.tsx

```diff
@@ -0,0 +1,225 @@
+        const message = clientError?.response?.errors?.[0]?.message;
+        if (message) {
+          setSubmitWorkEmailHint(message);
+        } else {
+          displayToast(labels.error.generic);
+        }
+      },
+    },
+  );
+
+  const { mutate: removeUserCompany } = useMutation(
+    (email: string) =>
+      gqlClient.request(REMOVE_USER_COMPANY_MUTATION, { email }),
+    {
+      onSuccess: (_, email) => {
+        queryClient.setQueryData<UserCompany[]>(
+          generateQueryKey(RequestKey.UserCompanies, user),
+          (currentUserCompanies) => {
+            return currentUserCompanies?.filter(
+              (userCompany) => userCompany.email !== email,
+            );
+          },
+        );
+      },
+      onError: () => displayToast(labels.error.generic),
+    },
+  );
+
+  const onDeleteUserCompany = async (email) => {
+    if (
+      await showPrompt({
+        title: 'Delete company verification',
+        description: 'Are you sure you want to delete this verified company?',
+        okButton: {
+          title: 'Yes, delete',
+        },
+      })
+    ) {
+      removeUserCompany(email);
+    }
+  };
+
+  return (
+    <AccountContentSection
+      title="Verified company badge"
+      description={
+        !userCompanyVerified &&
+        'Verify your work email and get a verified company badge on your profile. We won’t require any ID or personal information, just a verification code to complete the process.'
+      }
+    >
+      {userCompanyVerified && (
+        <div className="mt-6 max-w-sm">
+          <ul>
+            {userCompanies.map((userCompany) => (
+              <li className="flex items-center gap-3" key={userCompany.email}>
+                <ProfilePicture
+                  user={{
+                    image: userCompany.company.image,
+                    id: userCompany.company.name,
+                  }}
+                  rounded="full"
+                />
+                <div className="flex-1">
+                  <Typography type={TypographyType.Callout}>
+                    {userCompany.company.name}
+                  </Typography>
+                  <Typography
+                    type={TypographyType.Footnote}
+                    color={TypographyColor.Tertiary}
+                  >
+                    {userCompany.email}
+                    <Typography
+                      className="ml-1"
+                      color={TypographyColor.StatusSuccess}
+                      tag={TypographyTag.Span}
+                    >
+                      Verified
+                    </Typography>
+                  </Typography>
+                </div>
+                <Button
+                  type="button"
+                  variant={ButtonVariant.Tertiary}
+                  onClick={() => onDeleteUserCompany(userCompany.email)}
+                  icon={<TrashIcon />}
+                />
+              </li>
+            ))}
+          </ul>
+        </div>
+      )}
```

Comment:
Why not have these and the ones below in separate components? Can be in the same file if it makes more sense but at least to separate concerns

Reply:
Can do :)

Reply:
@nensidosari I moved the 2 render components, the form itself is a issue due to re-render issues on forms we have.
EG you would type and it would stop on every key stroke 🥶